Liberty

pjbreaux opened the pull request:
@mattgreene 

#### What's this change do?
This PR is to merge liberty into master. Once that is done, I will create a branch for liberty and mitaka based on the tip of master. Both liberty and mitaka installations of these playbooks have been tested.

#### Any background context?

#### Where should the reviewer start?
Review the license, contributing doc, and README file, just to make sure it's fit for public consumption.

pjbreaux commented:
All yaml files have been reviewed by @dflanigan, but feel free to peruse them if you'd like

mattgreene commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,67 @@
+<!--
+Copyright 2016 F5 Networks Inc.
```
Fix copyright to include 2017.

mattgreene commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,67 @@
+* Make sure you have tested your code because we are going to do that when when you make your PR.  You don't want _The Hat_ because your request fails unit tests.
+*  Clean up your git history because no one wants to see 75 commits for one issue
+*  Use our [commit template](.git-commit-template.txt)
+*  Use our pull request template
```
Create the magic .github directory with the following files:
PULL_REQUEST_TEMPLATE.md
ISSUE_TEMPLATE.md

See the agent repo for samples.

mattgreene commented on README.rst:
```diff
@@ -0,0 +1,65 @@
+F5 OpenStack Ansible
+====================
+
+|slack badge|
```
I see the new ansible channel on F5OpenStack slack.  Are you also going to create a channel on F5OpenStackDev slack?

pjbreaux replied:
Done

mattgreene commented on playbooks/hosts:
```diff
@@ -0,0 +1,2 @@
+[controllers]
+10.190.4.147 ansible_ssh_private_key_file=/root/.ssh/id_rsa_testlab ansible_ssh_extra_args='-o StrictHostKeyChecking=no'
```
Remove the real IP address so someone doesn't accidentally clober this machine in the future.

mattgreene commented on playbooks/roles/configure_lbaasv2_agent/tasks/main.yml:
```diff
@@ -0,0 +1,60 @@
+# version: 2016-12-20
+---
```
Can you add a comment to this file simply to state that the agent ini contains many settings, but this file is starting with the most commonly used settings?  (and feel free to add more as required)

mattgreene commented on playbooks/roles/configure_lbaasv2_agent/vars/main.yml:
```diff
@@ -0,0 +1,2 @@
+---
+undercloud: true
```
We've been conflating over and under cloud with GRM or L2 Adjacent Mode.  Rename this with "f5_global_routed_mode" which can be use directly to populate the ini file.

mattgreene commented on playbooks/vars/agent_configuration_vars.yaml:
```diff
@@ -0,0 +1,6 @@
+# Enter the values for the agent configuration
+bigip_netloc: <bigip_ip_address_or_hostname>
+agent_ini_file: /etc/neutron/services/f5/f5-openstack-agent.ini
+undercloud: <true_or_false>     # Set to true by default. Set to false if using an undercloud deployment.
```
See my previous comment in playbooks/roles/configure_lbaasv2_agent/vars/main.yml.  In fact, is the main.yml file necessary if this setting exists here?

pjbreaux replied:
We can remove the file and require the user to set the value

mattgreene replied:
ok

mattgreene commented on playbooks/vars/agent_configuration_vars.yaml:
```diff
@@ -0,0 +1,6 @@
+bigip_netloc: <bigip_ip_address_or_hostname>
+agent_ini_file: /etc/neutron/services/f5/f5-openstack-agent.ini
+undercloud: <true_or_false>     # Set to true by default. Set to false if using an undercloud deployment.
+tunnel_types: <tunnel_type>     # vxlan, gre etc...
```
Rename to "advertised_tunnel_types" to match the same name in the ini file.

mattgreene commented on playbooks/agent_driver_deploy.yaml:
```diff
@@ -0,0 +1,44 @@
+  become_method: sudo
+  become: true
+  remote_user: '{{ remote_user }}'
+  roles:
```
We need to be able to locate the agent on a different host than the driver (neutron server).  Can we either split driver and agent apart into separate files, or somehow provide the 'roles' on a per host basis?

pjbreaux replied:
This just sounds like a different playbook. Would you like me to add a new one?

mattgreene replied:
I prefer separate playbooks for driver and agent.

pjbreaux replied:
Updating to add new playbooks for driver deployment, then a separate one for the agent deployment.

pjbreaux commented:
@mattgreene: I addressed your comments. I created two new playbooks. The deployment of the agent alone and of the driver alone. I kept the original playbook to deploy both as well. For each playbook, I add a vars file. One vars file would be more confusing to just deploy the agent, but we could do that, if you'd like.

mattgreene commented:
This code is fine for 'master', but when it splits into liberty and mitaka branches the vars file needs to reflect the correct package versions.  playbooks/vars/agent_deploy_vars.yaml refers to 9.1.0 (which is mitaka, though your PR origin branch is liberty).

pjbreaux commented:
Closing
